atlasdb-tests-shared/src/main/java/com/palantir/timelock/paxos/InMemoryTimelockServices.java

```diff
@@ -0,0 +1,156 @@
+    private static final int THREAD_POOL_SIZE = 128;
+    private static final int BLOCKING_TIMEOUT_MS = 60 * 800; // 0.8 mins to ms
+
+    private final TimeLockServices delegate;
+    private final TimeLockAgent timeLockAgent;
+
+    private InMemoryTimelockServices(TimeLockServices delegate, TimeLockAgent timeLockAgent) {
+        this.delegate = delegate;
+        this.timeLockAgent = timeLockAgent;
+    }
+
+    @Override
+    public void close() {
+        timeLockAgent.shutdown();
+    }
+
+    public static InMemoryTimelockServices create(TemporaryFolder tempFolder) {
+        return create(tryCreateSubFolder(tempFolder));
+    }
+
+    private static InMemoryTimelockServices create(File dataDirectory) {
+        PaxosInstallConfiguration paxos = PaxosInstallConfiguration.builder()
+                .dataDirectory(dataDirectory)
+                .leaderMode(PaxosLeaderMode.SINGLE_LEADER)
+                .isNewService(false)
+                .build();
+
+        ClusterInstallConfiguration cluster = ImmutableClusterInstallConfiguration.builder()
+                .enableNonstandardAndPossiblyDangerousTopology(true)
+                .build();
+        TimeLockInstallConfiguration install = TimeLockInstallConfiguration.builder()
+                .paxos(paxos)
+                .cluster(cluster)
+                .build();
+
+        ImmutableDefaultClusterConfiguration clusterConfig = ImmutableDefaultClusterConfiguration.builder()
+                .localServer("local")
+                .cluster(PartialServiceConfiguration.of(List.of("local"), Optional.empty()))
+                .build();
+        TimeLockRuntimeConfiguration runtime = ImmutableTimeLockRuntimeConfiguration.builder()
+                .clusterSnapshot(clusterConfig)
+                .build();
+
+        TimeLockAgent timeLockAgent = TimeLockAgent.create(
+                MetricsManagers.of(new MetricRegistry(), SharedTaggedMetricRegistries.getSingleton()),
+                install,
+                Refreshable.only(runtime), // This won't actually live reload.
+                runtime.clusterSnapshot(),
+                USER_AGENT,
+                THREAD_POOL_SIZE,
+                BLOCKING_TIMEOUT_MS,
+                _unused -> {},
+                Optional.empty(),
+                OrderableSlsVersion.valueOf("0.0.0"),
+                ObjectMappers.newServerObjectMapper(),
+                () -> System.exit(0));
+
+        TimeLockServices services = timeLockAgent.createInvalidatingTimeLockServices("client");
+
+        // Wait for leadership
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(30L))
+                .pollInterval(Duration.ofMillis(50))
+                .ignoreExceptions()
+                .until(() -> services.getTimestampService().getFreshTimestamp() > 0);
+
+        return new InMemoryTimelockServices(services, timeLockAgent);
+    }
```

Comment:
Is there a reason we fixing this delegate to a single client? Like I think there could be a test where we have two different clients talking to a TL service.

Reply:
We could do that if we wanted to - we'd just need to reorganise the class a bit to maintain a `Map<Client, TimelockAgent`, and I haven't yet seen a place where we'd want that